app/src/main/java/ru/practicum/android/diploma/ui/vacancy/VacancyFragment.kt

```diff
@@ -1,15 +1,146 @@
 package ru.practicum.android.diploma.ui.vacancy
 
+import android.os.Bundle
+import android.text.Html
 import android.view.LayoutInflater
+import android.view.View
 import android.view.ViewGroup
+import androidx.core.os.bundleOf
+import androidx.navigation.fragment.findNavController
+import com.bumptech.glide.Glide
+import org.koin.android.ext.android.inject
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import ru.practicum.android.diploma.R
+import ru.practicum.android.diploma.app.NavigationJobApp.Companion.applicationContext
+import ru.practicum.android.diploma.common.AppConstants.EMPTY_PARAM_VALUE
+import ru.practicum.android.diploma.data.formatter.SalaryFormatter
 import ru.practicum.android.diploma.databinding.FragmentVacancyBinding
+import ru.practicum.android.diploma.domain.models.VacancyDetails
+import ru.practicum.android.diploma.domain.state.VacancyDetailsState
 import ru.practicum.android.diploma.util.BindingFragment
+import ru.practicum.android.diploma.util.ImageAndTextHelper
+import ru.practicum.android.diploma.util.invisible
+import ru.practicum.android.diploma.util.visible
 
 class VacancyFragment : BindingFragment<FragmentVacancyBinding>() {
 
+    private val viewModel: VacancyViewModel by viewModel()
+    private val imageAndTextHelper: ImageAndTextHelper by inject()
+    private val salaryFormatter = SalaryFormatter(applicationContext())
+
     override fun createBinding(
         inflater: LayoutInflater,
         container: ViewGroup?
     ) = FragmentVacancyBinding.inflate(inflater, container, false)
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val vacancyId = requireArguments().getString(VACANCY_DETAILS) ?: ""
```

Comment:
```suggestion
        val vacancyId = requireArguments().getString(VACANCY_DETAILS).orEmpty()
```

Аналогично в других местах

Reply:
Done